react-graph/src/PointNode.tsx

```diff
@@ -205,6 +214,15 @@ export class PointNode {
         return children;
     }
 
+    private trimTree(): void {
+        const fullData = this.GetFullData();
+        if (fullData.length < MaxTotalPoints) return
+
+        const halfIndex = Math.floor(fullData.length / 2);
+        const trimmedData = fullData.slice(halfIndex);
+        this.initializeNode(trimmedData);
```

Comment:
Also this Logic may be confusing. If we do have more than `MaxTotalPoints` we may want to remove Points at the end, But I am pretty sure just tossing half the data will result in some unexpected behaviour